flink-table/flink-table-runtime/src/test/java/org/apache/flink/table/runtime/operators/join/stream/StreamingMiniBatchJoinOperatorTest.java

```diff
@@ -849,6 +851,46 @@ public void testInnerJoinWithNoUniqueKeyWithinBatch() throws Exception {
                         "4 Bellevue Drive, Pottstown, PB 19464"));
     }
 
+    @Tag("miniBatchSize=4")
+    @Test
+    public void testInnerJoinWithNoUniqueKeyHashCollision(TestInfo testInfo) throws Exception {
```

Comment:
[FLINK-35184.patch](https://github.com/apache/flink/files/15074895/FLINK-35184.patch)
Could you make some adjustments to this test? Currently, all tests are being conducted using the same schema. If a new schema has been added, could you refactor this part?  The attachment contains my modifications and welcome to your feedback.

Reply:
ok, agree, it would be better to refactor this. Used Extractors for this purpose.